File: example_bot.nim
import botdev, tables

var bot = BotDev(players: initTable[string, Player]())

bot.addPlayer("Alice")
bot.addPlayer("Bob")

registerCommand("attack", proc(player: var Player, args: seq[string]): string =
  return player.name & " attacks!"
)

registerCommand("chat", proc(player: var Player, args: seq[string]): string =
  return player.name & " says: " & aiResponse(args.join(" "))
)

registerCommand("collect", proc(player: var Player, args: seq[string]): string =
  let items = @["Sword", "Potion", "Shield", "Gem"]
  let item = items[rand(items.len)]
  addItem(player, item)
  return player.name & " collected " & item
)

echo execute(bot, "attack", "Alice", @[])
echo execute(bot, "chat", "Bob", @["Hello!"])
echo execute(bot, "collect", "Alice", @[])

File: example_bot.m
#import <Foundation/Foundation.h>
#import "BotDev.h"

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        BotDev *bot = [[BotDev alloc] init];
        
        [bot registerCommand:@"attack" withBlock:^NSString* (NSString* player, NSArray* args) {
            return [NSString stringWithFormat:@"%@ attacks!", player];
        }];
        
        [bot registerCommand:@"chat" withBlock:^NSString* (NSString* player, NSArray* args) {
            NSString *msg = [args componentsJoinedByString:@" "];
            NSString *resp = [bot aiResponse:msg];
            return [NSString stringWithFormat:@"%@ says: %@", player, resp];
        }];
        
        [bot onEvent:@"player_joined" block:^(NSArray *args) {
            NSLog(@"[EVENT] Welcome %@!", args[0]);
        }];
        
        [bot addPlayer:@"Alice"];
        [bot addPlayer:@"Bob"];
        
        NSLog(@"%@", [bot executeCommand:@"attack" forPlayer:@"Alice" args:@[]]);
        NSLog(@"%@", [bot executeCommand:@"chat" forPlayer:@"Bob" args:@[@"Hello!"]]);
    }
    return 0;
}
